doc/developer-guide/api/functions/TSUrlHostGet.en.rst

```diff
@@ -52,6 +52,10 @@ buffers. The URL functions can create, copy, retrieve or delete entire URLs,
 and retrieve or modify parts of URLs, such as their host, port or scheme
 information.
 
+    TSMBufferCreate(...)
+    TSUrlParse(...)
+    TSUrlHostGet(...)
```

Comment:
I don't  understand what these three line communicate.

Reply:
The current issue at hand indicates the inclusion of a specific note in the document. The note emphasizes that the function TSUrlHostGet can only be invoked subsequent to the TS_HTTP_POST_REMAP_HOOK stage. However, during the TS_HTTP_READ_REQUEST_HDR_HOOK phase, one can make use of the TSHttpHdrHostGet function to retrieve the host information if required.

In response to this issue, jpeach suggested incorporating a condition for calling TSUrlHostGet into the document, as expressed in #5742. According to jpeach, TSUrlHostGet receives a URL from a marshal buffer, and as long as the buffer contains the necessary host portion, the function should function correctly.

To implement this, the following steps can be taken:

1. Create a TSMBuffer instance using the TSMBufferCreate() function.
2. Parse the URL using the TSUrlParse() function.
3. Retrieve the host using the TSUrlHostGet() function.
Therefore, to address the issue, I modified the code as follows:

TSMBufferCreate(...);
TSUrlParse(...);
TSUrlHostGet(...);

By making these modifications, the code will adhere to the suggested approach while maintaining a professional and easily understandable format for your reviewer.

Reference :- https://github.com/apache/trafficserver/issues/5742

Reply:
Hmm, I'm still confused.  I think you are touching on an entirely separate issue, that the **Description** obscures the fact that *offset* refers to a URL.  But TSUrlParse() is *not* the only way to get a TSMLoc that refers to a URL.  TSHttpHdrUrlGet() and TSHttpTxnPristineUrlGet() also provide a TSMLoc value that refers to a URL.

Also I don't understand why you prefer a terse outline of a code example, rather than a more complete code example.

Reply:
Hello @ywkaras, I'm seeking your guidance to resolve issue #5742. I apologize for any confusion caused and greatly appreciate your expertise. Could you please clarify the specific documentation modifications needed to address the confusion surrounding the "offset" referring to a URL? Any additional instructions or examples related to the necessary adjustments would be immensely helpful. I'm committed to implementing the required changes and value your guidance accurately. Thank you for your time and support. What exactly needs to be done to resolve #5742?

Reply:
To understand the use of URL TSMLoc values, I think your best next step is to grep/search for the use of the relevant functions in the code.

The issue is related to which hooks ( https://docs.trafficserver.apache.org/developer-guide/plugins/hooks-and-transactions/index.en.html#http-transaction-state-diagram ) an invoked continuation function make call TSUrlGetHost(). It will take lengthy study of ATS for all this to be clear to you.